include/swift/AST/DiagnosticsSema.def

```diff
@@ -2814,7 +2814,7 @@ WARNING(expression_unused_result_call,none,
 WARNING(expression_unused_result_operator,none,
         "result of operator %0 is unused", (DeclName))
 WARNING(expression_unused_result_unknown, none,
-        "result of call is unused, but produces %0", (Type))
+        "result of call to closure returning %0 is unused", (Type))
```

Comment:
This is not quite what I was pointing out - not everything this diagnoses is a call to a closure, nor is everything it diagnoses a call to a function.  So you have two options:

1) Conditionalize this diagnostic on whether the called value is a closure or a function.
2) Just delete it and use `expression_unused_result_call` instead.

I really recommend #2.  This diagnostic seems genuinely superfluous.

Reply:
@CodaFi Thanks for the context. I thought it is just for closure and will not prefer conditionalizing it. I will update it then.